evaluation.py

```diff
@@ -135,3 +137,38 @@ def load_and_preprocess_img(img_path, num_hg_blocks, x=None, y=None, w=None, h=N
     y_batch = [np.zeros((1, *(OUTPUT_DIM), NUM_COCO_KEYPOINTS)) for _ in range(num_hg_blocks)]
 
     return X_batch, y_batch
+
+    # Resources for heatmaps to keypoints
+    # https://github.com/yuanyuanli85/Stacked_Hourglass_Network_Keras/blob/eddf0ae15715a88d7859847cfff5f5092b260ae1/src/eval/heatmap_process.py#L5
+    # https://github.com/david8862/tf-keras-stacked-hourglass-keypoint-detection/blob/56707252501c73b2bf2aac8fff3e22760fd47dca/hourglass/postprocess.py#L17
+   
+    ### Returns np array of predicted keypoints from one image's heatmaps
+    def heatmaps_to_keypoints(self, heatmaps, threshold=1e-2):
+        keypoints = list()
+        for i in range(NUM_COCO_KEYPOINTS):
+            hmap = heatmaps[:,:,i]
+            # Do a heatmap blur with gaussian_filter
+            hmap = gaussian_filter(hmap, HEATMAP_SIGMA)
+
+            # Resize heatmap from Output DIM to Input DIM
+            resized_hmap = cv2.resize(hmap, INPUT_DIM, interpolation = cv2.INTER_AREA)
+
+            # Get peak point (brightest area) in heatmap with 3x3 max filter
+            peaks = self._non_max_supression(resized_hmap, windowSize=3, threshold=1e-2)
+
+            # Choose the max point in heatmap (we only pick 1 keypoint in each heatmap)
+            # and get its coordinates and confidence
+            y, x = np.where(peaks == peaks.max())
+            if len(x) > 0 and len(y) > 0:
+                keypoints.append((int(x[0]), int(y[0]), peaks[y[0], x[0]]))
```

Comment:
what is the additional param here? a confidence level of sorts?

Reply:
yeah so this I took from the other repo it's a confidence level, not sure if we want to use this for any reason or just append a 1 or 2 as the v number? (this doesn't get used by the OKS evaluation)

Reply:
yeah let's keep it for now. we can remove the kp later if it's not confident enough